lib/orocos/name_service.rb

```diff
@@ -650,7 +645,16 @@ def deregister(name)
             #(see NameServiceBase#validate)
             def validate
                 CORBA.refine_exceptions("corba naming service #{ip}") do
-                    do_validate
+                    begin
+                        do_validate
+                    rescue Orocos::ComError => e
+                        CORBA.warn "Name service is unreachable: #{e.message}\n"
+                        puts "You can try to fix this manually by restarting the nameserver:"
+                        puts "    sudo /etc/init.d/omniorb4-nameserver stop"
+                        puts "    sudo rm -f /var/lib/omniorb/*"
+                        puts "    sudo /etc/init.d/omniorb4-nameserver start"
+                        raise Orocos::ComError, e.backtrace
```

Comment:
This actually loses the exception message. Re-raising an exception in Ruby is just done by `raise` without arguments

Reply:
Thank you for the info. Edited the commit accordingly.